vpr/src/route/rr_graph_storage.h

```diff
@@ -0,0 +1,578 @@
+#ifndef _RR_NODE_STORAGE_
+#define _RR_NODE_STORAGE_
+
+#include <exception>
+
+#include "rr_node_fwd.h"
+#include "rr_graph2.h"
+#include "vtr_log.h"
+#include "vtr_memory.h"
+#include "vpr_utils.h"
+
+/* Main structure describing one routing resource node.  Everything in       *
+ * this structure should describe the graph -- information needed only       *
+ * to store algorithm-specific data should be stored in one of the           *
+ * parallel rr_node_* structures.                                            *
+ *                                                                           *
+ * This structure should **only** contain data used in the inner loop of the *
+ * router.  This data is consider "hot" and the router performance is        *
+ * sensitive to layout and size of this "hot" data.
+ *
+ * Cold data should be stored seperately in t_rr_graph_storage.              *
+ *                                                                           *
+ * xlow, xhigh, ylow, yhigh:  Integer coordinates (see route.c for           *
+ *       coordinate system) of the ends of this routing resource.            *
+ *       xlow = xhigh and ylow = yhigh for pins or for segments of           *
+ *       length 1.  These values are used to decide whether or not this      *
+ *       node should be added to the expansion heap, based on things         *
+ *       like whether it's outside the net bounding box or is moving         *
+ *       further away from the target, etc.                                  *
+ * type:  What is this routing resource?                                     *
+ * cost_index: An integer index into the table of routing resource indexed   *
+ *             data t_rr_index_data (this indirection allows quick dynamic   *
+ *             changes of rr base costs, and some memory storage savings for *
+ *             fields that have only a few distinct values).                 *
+ * capacity:   Capacity of this node (number of routes that can use it).     *
```

Comment:
Add rc_index comment: flyweight index for R and C values for this node.

Reply:
Done